🐛  (kustomize/v1 and kustomize/v2-alpha) : ComponentConfig scaffolds should not be done by default (#2826)
kubernetes-sigs/kubebuilder: docs/book/src/component-config-tutorial/api-changes.md

With that out of the way, we can get on to defining our new config!

[tutorial-source]: https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/component-config-tutorial/testdata/project

> ## If you miss flag `--component-config` during the `kubebuilder init` command, then add it manually as shown below.

Reviewer: @laxmikantbpandhare see that we have an subsection `Setting up an exising project` where add the steps about what you need to do if you do not scaffold the project with the flag:
https://github.com/kubernetes-sigs/kubebuilder/blob/9dcd389aa9e2386af20b08e045d0334e788c768b/docs/book/src/component-config-tutorial/api-changes.md#setting-up-an-exising-project

So, ihmo it needs to be +1 step under this subsection instead of a new one.

Reviewer: Also, we need to add here the step to add to the file: 

```
generatorOptions:
  disableNameSuffixHash: true

configMapGenerator:
- name: manager-config
  files:
  - controller_manager_config.yaml
```

Author: Updated file and made above as subsection in `Setting up an existing project`. Also, I added the above missing content.

With that out of the way, we can get on to defining our new config!

[tutorial-source]: https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/component-config-tutorial/testdata/project

> Add below content to this file ->`/config/manager/controller_manager_config.yaml`

Reviewer: ```suggestion
Create the file "/config/manager/controller_manager_config.yaml" with the following content:
```

Author: Updated.

```yaml
apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
kind: ControllerManagerConfig
health:
  healthProbeBindAddress: :8081
metrics:
  bindAddress: 127.0.0.1:8080
webhook:
  port: 9443
leaderElection:
  leaderElect: true
  resourceName: ecaf1259.tutorial.kubebuilder.io
# leaderElectionReleaseOnCancel defines if the leader should step down volume
# when the Manager ends. This requires the binary to immediately end when the
# Manager is stopped, otherwise, this setting is unsafe. Setting this significantly
# speeds up voluntary leader transitions as the new leader don't have to wait
# LeaseDuration time first.
# In the default scaffold provided, the program ends immediately after
# the manager stops, so would be fine to enable this option. However,
# if you are doing or is intended to do any operation such as perform cleanups
# after the manager stops then its usage might be unsafe.
# leaderElectionReleaseOnCancel: true
```

> Add below content to this file ->`/config/manager/kustomization.yaml`

Reviewer: ```suggestion
Update the file `/config/manager/kustomization.yaml` by adding at the botton the following content:
```

Author: Updated.

```yaml
generatorOptions:
  disableNameSuffixHash: true

configMapGenerator:
- name: manager-config
  files:
  - controller_manager_config.yaml

> ```

Reviewer: ```suggestion
```

Update the `default/kustomization.yaml` by adding under the `patchesStrategicMerge:` the following patch:

```yaml

# Mount the controller config file for loading manager configurations
# through a ComponentConfig type
- manager_config_patch.yaml
```

```

```

> ```

Reviewer: ```suggestion
```

Update the `default/kustomization.yaml` by adding under the `patchesStrategicMerge:` the following patch:

```yaml

# Mount the controller config file for loading manager configurations
# through a ComponentConfig type
- manager_config_patch.yaml
```

```

Author: Yes, I updated the docs section.

[tutorial-source]: https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/component-config-tutorial/testdata/project

Create the file `/config/manager/controller_manager_config.yaml` with the following content:

```yaml
apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
kind: ControllerManagerConfig
health:
  healthProbeBindAddress: :8081
metrics:
  bindAddress: 127.0.0.1:8080
webhook:
  port: 9443
leaderElection:
  leaderElect: true
  resourceName: ecaf1259.tutorial.kubebuilder.io
# leaderElectionReleaseOnCancel defines if the leader should step down volume
# when the Manager ends. This requires the binary to immediately end when the
# Manager is stopped, otherwise, this setting is unsafe. Setting this significantly
# speeds up voluntary leader transitions as the new leader don't have to wait
# LeaseDuration time first.
# In the default scaffold provided, the program ends immediately after
# the manager stops, so would be fine to enable this option. However,
# if you are doing or is intended to do any operation such as perform cleanups
# after the manager stops then its usage might be unsafe.
# leaderElectionReleaseOnCancel: true
```

> Update the file `/config/manager/kustomization.yaml` by adding at the botton the following content:

Reviewer 2: Typo:
```suggestion
Update the file `/config/manager/kustomization.yaml` by adding at the bottom the following content:
```

Author: Updated.
